### PR TITLE
Extract header into its own interface

### PR DESCRIPTION
**Pull request by victornoel**
For #77

### CODE REVIEW

**Comment by victornoel** [comment]
@0crat status

**Comment by victornoel** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/rpm-adapter/issues/102#issuecomment-612670996))

@victornoel This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#102](https://github.com/artipie/rpm-adapter/issues/102) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Frpm-adapter%23102%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@g4s8 I'm not sure why but this PR didn't go into scope

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/rpm-adapter/issues/102#issuecomment-612769159))

@g4s8 This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#102](https://github.com/artipie/rpm-adapter/issues/102) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Frpm-adapter%23102%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat in

**Comment by 0crat** [comment]
This pull request [#102](https://github.com/artipie/rpm-adapter/issues/102) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/64186746-3609-4a41-9aac-4251103f0b6d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/pkg/FilePackage.java [hunk 1]
```diff
@@ -183,4 +158,53 @@ public String href() {
...
+        /**
+         * Native header entry.
+         */
+        private final Optional<AbstractHeader.Entry<?>> entry;
+
+        /**
+         * Ctor.
+         * @param entry Native header entry
+         */
+        EntryHeader(final Optional<AbstractHeader.Entry<?>> entry) {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/pkg/FilePackage.java:178]: @victornoel let's add a ctor that takes `AbstractHeader` and wraps it to optional to make it more user friendly. Than we can remove ugly `Optional.ofNullable` from `FilePackage.header()` method.

File: src/main/java/com/artipie/rpm/pkg/FilePackage.java [hunk 2]
```diff
@@ -183,4 +158,53 @@ public String href() {
             };
         }
     }
+
+    /**
+     * {@link AbstractHeader.Entry} based MetaHeader.
+     *
+     * @since 1.0
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/pkg/FilePackage.java:165]: @victornoel since `0.6.3`

File: src/main/java/com/artipie/rpm/pkg/Package.java [hunk 3]
```diff
@@ -107,4 +80,36 @@
          */
         int[] range();
     }
+
+    /**
+     * Package metadata header.
+     * @since 1.0
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/pkg/Package.java:86]: @victornoel since `0.6.3`

**Comment by victornoel** [comment]
@Vatavuk thank you, it is done

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find not only cosmetic issues, but rather design problems during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**